kernel/userspace.c

```diff
@@ -94,6 +98,185 @@ struct perm_ctx {
 	struct k_thread *parent;
 };
 
+#ifdef CONFIG_DYNAMIC_OBJECTS
+struct dyn_obj {
+	struct _k_object kobj;
+	struct rbnode node; /* must be immediately before data member */
+	u8_t data[]; /* The object itself */
+};
+
+struct visit_ctx {
+	_wordlist_cb_func_t func;
+	void *original_context;
+};
+
+extern struct _k_object *_k_object_gperf_find(void *obj);
+extern void _k_object_gperf_wordlist_foreach(_wordlist_cb_func_t func,
+					     void *context);
+
+static int node_lessthan(struct rbnode *a, struct rbnode *b);
+
+static struct rbtree obj_rb_tree = {
+	.lessthan_fn = node_lessthan
+};
+
+/* TODO: incorporate auto-gen with Leandro's patch */
+static size_t obj_size_get(enum k_objects otype)
```

Comment:
This is another function which have to be kept in sync with k_objects enum.  Could you please mention that fact in a comment around k_objects declaration?

Reply:
This will be taken care of -- there's a pending PR (#6930) that will automatically generate a bunch of things, including enum k_objects itself, and the string representation.  As soon as that's merged, I can change the script to also generate the contents of this function.